File: src/java/principles_systems_architecture/chapter1/section2/ShippingCost.java
package principles_systems_architecture.chapter1.section2;

class ShippingCost {
    static final int MINIMUM_FOR_FREE = 3000;
    static final int COST = 500;

    final int basePrice;

    ShippingCost(int basePrice) {
        this.basePrice = basePrice;
    }

    int amount() {
        if (this.basePrice < MINIMUM_FOR_FREE) return COST;
        return 0;
    }

    @Override
    public String toString() {
        return String.format("ShippingCost(basePrice = %s)", this.basePrice);
    }
}

File: src/clj/principles_systems_architecture/chapter1/section2/shipping_cost.clj
(ns principles-systems-architecture.chapter1.section2.shipping-cost
  (:require [schema.core :as s]))

(s/def minimum-for-free :- Long
  3000)

(s/def cost :- Long
  500)

(s/defn amount :- Long
  [base-price :- Long]
  (if (< base-price minimum-for-free)
    cost
    0))
